src/frontends/onnx/frontend/src/ops_bridge.cpp

```diff
@@ -252,41 +246,41 @@ OperatorSet OperatorsBridge::_get_operator_set(const std::string& domain, std::i
     return result;
 }
 
-bool OperatorsBridge::_is_operator_registered(const std::string& name,
-                                              std::int64_t version,
-                                              const std::string& domain) {
-    std::lock_guard<std::mutex> guard(lock);
+bool OperatorsBridge::is_operator_registered(const std::string& name,
+                                             int64_t version,
+                                             const std::string& domain) const {
     // search for domain
-    auto dm_map = m_map.find(domain);
+    const auto dm_map = m_map.find(domain);
     if (dm_map == std::end(m_map)) {
         return false;
     }
     // search for name
-    auto op_map = dm_map->second.find(name);
+    const auto op_map = dm_map->second.find(name);
     if (op_map == std::end(dm_map->second)) {
         return false;
     }
 
-    if (find(version, op_map->second) != std::end(op_map->second)) {
-        return true;
-    } else {
-        return false;
+    return find(version, op_map->second) != std::end(op_map->second);
+}
+
+void OperatorsBridge::overwrite_operator(const std::string& name, const std::string& domain, Operator fn) {
+    const auto domain_it = m_map.find(domain);
+    if (domain_it != m_map.end()) {
+        auto& domain_opset = domain_it->second;
+        domain_opset[name].clear();
     }
+    register_operator(name, 1, domain, std::move(fn));
```

Comment:
The previous code was a bit redundant. This version first removes all existing implementations of an operator(the clear method above) and then registers one mapping function for version 1. This will ensure fallback for all higher versions to the implementation being registered here.